src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.SlhDsa.cs

```diff
@@ -100,26 +100,17 @@ protected override bool Sign(
                 // not enforce them here. It is up to the caller to choose an appropriate hash.
 
                 signatureParameters = null;
+                signatureAlgorithm = _signatureAlgorithm;
```

Comment:
You've moved an assignment of this out to earlier than it used to be.  Do any of the callers ever use this out in a shared memory location (i.e. a static field)?

It's generally best to only ever have one write to an out, just in case they become "side effect visible"; and this change makes signature failure go from non-null back to null.

Reply:
Oops, I'll remove the other `null` assignment below that I missed. The other algos also follow this pattern of assignment.